File: conformance/run-1/components/flow.ts
import { CapabilityError, handle } from '../../../packages/flow-sdk/src/index'

await handle(async (run) => {
  const researchPromise = run.runChildFlow({
    operationId: 'research:1',
    slot: 'research',
    intent: 'Find a useful comparison target.',
    input: run.input,
  })

  const storedPromise = run.callCapability({
    operationId: 'store:1',
    slot: 'artifacts',
    method: 'write',
    input: { source: 'research' },
  })

  const [research, stored] = await Promise.all([researchPromise, storedPromise])
  let missing: string | null = null

  try {
    await run.callCapability({
      operationId: 'missing:1',
      slot: 'artifacts',
      method: 'read',
      input: { uri: 'artifact://missing' },
    })
  } catch (error) {
    if (!(error instanceof CapabilityError)) throw error
    missing = error.errorName
  }

  return {
    outcome: 'done',
    output: { research, stored, missing },
  }
})

File: conformance/run-1/components/flow.py
from jiggy.flow import CapabilityError, handle


async def run(context):
    import asyncio

    research_task = asyncio.create_task(
        context.run_child_flow(
            operation_id="research:1",
            slot="research",
            intent="Find a useful comparison target.",
            input=context.input,
        )
    )
    stored_task = asyncio.create_task(
        context.call_capability(
            operation_id="store:1",
            slot="artifacts",
            method="write",
            input={"source": "research"},
        )
    )

    research, stored = await asyncio.gather(research_task, stored_task)
    missing = None

    try:
        await context.call_capability(
            operation_id="missing:1",
            slot="artifacts",
            method="read",
            input={"uri": "artifact://missing"},
        )
    except CapabilityError as error:
        missing = error.error_name

    return {
        "outcome": "done",
        "output": {
            "research": research,
            "stored": stored,
            "missing": missing,
        },
    }


handle(run)
